pkg/action/executor/resize_container_util.go

```
@@ -202,56 +175,116 @@ func genMemoryQuantity(newValue float64) (resource.Quantity, error) {
 	return result, nil
 }
 
-func resizeContainer(client *kclient.Clientset, pod *k8sapi.Pod, spec *containerResizeSpec, retryNum int) error {
+// Resize pod in three steps:
+//   step1: create a clone pod of the original pod (without labels), with new resource limits/requests;
+//   step2: delete the orginal pod;
+//   step3: add the labels to the cloned pod;
+func resizeContainer(client *kclient.Clientset, tpod *k8sapi.Pod, spec *containerResizeSpec, retryNum int) (*k8sapi.Pod, error) {
 	index := spec.Index
-	id := fmt.Sprintf("%s/%s-%d", pod.Namespace, pod.Name, index)
+	id := fmt.Sprintf("%s/%s-%d", tpod.Namespace, tpod.Name, index)
 	glog.V(2).Infof("begin to resize Pod container[%s].", id)
 
+	podClient := client.CoreV1().Pods(tpod.Namespace)
+	if podClient == nil {
+		err := fmt.Errorf("cannot get Pod client for nameSpace:%v", tpod.Namespace)
+		glog.Error(err)
+		return nil, err
+	}
+
 	// 0. get the latest Pod
 	var err error
-	if pod, err = autil.GetPod(client, pod.Namespace, pod.Name); err != nil {
+	var pod *k8sapi.Pod
+	if pod, err = podClient.Get(tpod.Name, metav1.GetOptions{}); err != nil {
 		glog.Errorf("Failed to get latest pod %v: %v", id, err)
-		return err
+		return nil, err
+	}
+	labels := pod.Labels
+
+	// 1. create a clone pod
+	npod, changed, err := createResizePod(client, pod, spec)
+	if err != nil {
+		glog.Errorf("resizeContainer failed[%s]: failed to create a resized pod: %v", id, err)
+		return nil, err
 	}
 
-	// 1. copy the original pod
+	if !changed {
+		glog.Warningf("resizeContainer aborted[%s]: no need do resize container.", id)
+		return nil, nil
```

If it's not changed, do we want to return an error for the resize action?